docs/descriptor.rst

```diff
@@ -0,0 +1,210 @@
+Image descriptor file
+=====================
+
+Most important, user facing part of Concreate is the descriptor file. We use
+`YAML <http://yaml.org/>`_ format which is easy to read and edit by humans but still machine
+parseable.
+
+.. note::
+
+    By convention we use the ``image.yaml``  file name for the image descriptor.
+
+Below you can find description of image descriptor sections.
+
+General information
+-------------------
+
+.. glossary::
+
+    ``name`` (required)
+        Image name without the registry part.
+
+    ``version`` (required)
+        Version of the image.
+
+    ``from`` (required)
+        Base image of your image.
+
+    ``schema_version`` (required)
+        Here you specify the schema version of the descriptor. This influences what versions of Concreate are able to parse it.
+
+    ``description``
+         Short summary of the image -- also used to create the ``description`` label.
+
+    ``user``
+        The user that should be used to launch the main process in the container. Can be name or uid. See `OCI spec <https://github.com/opencontainers/image-spec/blob/master/config.md#properties>`_ for more information.
+
+.. code:: yaml
+
+    name: "jboss-eap-7/eap70-openshift"
+    version: "1.4"
+    from: "jboss-eap-7-tech-preview/eap70:1.2"
+    description: "Red Hat JBoss Enterprise Application 7.0 - An application platform for hosting your apps that provides an innovative modular, cloud-ready architecture, powerful management and automation, and world class developer productivity."
+    user: "alice"
+
+Labels
+------
+
+Every image can include labels. Concreate makes it easy to do so with the ``labels`` section.
+
+.. code:: yaml
+
+    labels:
+        - name: "io.k8s.description"
+          value: "Platform for building and running JavaEE applications on JBoss EAP 7.0"
+        - name: "io.k8s.display-name"
+          value: "JBoss EAP 7.0"
+
+Environment variables
+---------------------
+
+Similar to labels -- we can specify environment variables that should be
+present in the container after running the image. We provide ``envs``
+section for this.
+
+Environment variables can be divided into two types:
+
+1. **Information environment variables** -- these are set and available in
```

Comment:
I would just skip the first sentence here and start with. These env... variables provide information... - maybe mention docker inspect to see them somewhere?

Reply:
Fro now on we will make the tool not tied to any implementation, including Docker.